rust_src/src/process.rs

```diff
@@ -537,4 +538,24 @@ pub fn list_system_processes_rust() -> LispObject {
     unsafe { list_system_processes() }
 }
 
+/// Interrupt process PROCESS
+/// PROCESS may be a process, a buffer, or the name of a process or buffer.
+/// No arg or nil means current buffer's process.
+/// Second arg CURRENT-GROUP non-nil means send signal to
+/// the current process-group of the process's controlling terminal
+/// rather than to the process's own process group.
+/// If the process is a shell, this means interrupt current subjob
+/// rather than the shell.
+
+/// If CURRENT-GROUP is `lambda', and if the shell owns the terminal,
+/// don't send the signal.
+
+/// This function calls the functions of `interrupt-process-functions' in
+/// the order of the list, until one of them returns non-`nil'.  */)
+#[lisp_fn(name="interrupt-process")]
```

Comment:
The only reason to use `name` is if the Lisp DEFUN has the same name as an existing function. In the C code name in the `DEFUN` macro is changed from `foo-bar` to `Ffoo_bar`. In Rust we define `foo_bar` and a macro creates a wrapper function called `Ffoo_bar` for us.

I do not see another `interrupt_process` so there is no need to name the function `_rust`. Also, we are trying to transition to using `_lisp` for the name of functions that are interfaces to the DEFUNs and `_rust` for internal functions that need to be differentiated from C or DEFUN code.